In [1]:
defaultTime:1000

(defaultTime:1000)

In [2]:
import : dir co : up : up : .

In [3]:
count : in : commutative.C0-S6.candidates

27

In [4]:
count : in : commutative.C2-S6.candidates 

977

In [5]:
tab : first 20 : in : commutative.C2-S6.candidates 

 (:is {B null}) 
 (:null {nif B}) 
 (:null {1 : B}) 
 (:null {B : pass}) 
 (:const {prod B}) 
 (:if {B : right}) 
 (:if {B : plus}) 
 (:null {if B}) 
 (:has prod) 
 (:get {left B}) 
 (:has {isnt B}) 
 (:{const : B} is) 
 (:get {B : b}) 
 (:null {0 : B}) 
 (:is null) 
 (:if {B get}) 
 (:is {hasnt B}) 
 (:null {B : put}) 
 (:const {B : star}) 
 (:has {const : B}) 


## How many of the 977 commutative candidates are also symmetric? 

In [6]:
Let Class : classify S6?  : (in : commutative.C0-S6.candidates) 
#                           (in : commutative.C2-S6.candidates) 

In [7]:
count : Class?

9

In [8]:
tab : Class?

 (bin 10:(:const) (:get) (:has) (:head) (:is) (:nif) (:nth) (:nth1) (:null) (:sum)) 
 (bin 1:(:count)) 
 (bin 3:(:float_max) (:float_min) (:float_sort)) 
 (bin 1:(:float_prod)) 
 (bin 1:(:float_sum)) 
 (bin 6:(:int_max) (:int_min) (:int_sort) (:nat_max) (:nat_min) (:nat_sort)) 
 (bin 2:(:int_prod) (:nat_prod)) 
 (bin 2:(:int_sum) (:nat_sum)) 
 (bin 1:(:text_sort)) 


In [11]:
out R1 3600 : dom 100 : property commutative S6? : in : commutative.C0-S6.candidates 

In [ ]:
out R2 3600 : dom 100 : property commutative S6? : in : commutative.C2-S6.candidates 